### ISSUE TITLE
[java] ArrayIsStoredDirectly false positive with field access

### ISSUE THREAD

**Issue by anicoara** [pmd/pmd#1185]
**Affects PMD Version:** 6.4.0

**Rule:** [ArrayIsStoredDirectly](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#arrayisstoreddirectly)

**Description:** False positive when passing field, that shares the same name as input parameter, to constructor of different object.

**Code Sample demonstrating the issue:**

```java
class TestArrayIsStoredDirectly {
    private final boolean[] a;
    private final foo b;

    private TestArrayIsStoredDirectly(boolean[] a) {
        this.a = null;
        this.a = a.clone(); // no violation, it doesn't matter what the state of this.a is
        this.b = new TestArrayIsStoredDirectly(a); // no violation
        this.b = new TestArrayIsStoredDirectly(this.a); // false positive violation here
    }
}
```

Rule Property: allowPrivate=false

**Running PMD through:** *[CLI | Maven]*
